node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -143,19 +142,21 @@ func (ms *P2PMessageService) setupDht(bootPeers []string) {
 	addrs, err := peer.AddrInfoToP2pAddrs(&peerInfo)
 	ms.checkError(err)
 	ms.MultiAddr = addrs[0].String()
-	ms.logger.Debug().Msgf("libp2p node address: %s", ms.MultiAddr)
+	ms.logger.Debug().Msgf("libp2p node multiaddrs: %v", addrs)
+
+	// Setup notifications so we exchange nitro signing addresses when connected
+	n := &NetworkNotifiee{ms: ms}
```

Comment:
Instead of wrapping a message service in a `NetworkNotifiee` struct, the message service itself could just implement the `Notifee` interface. That would avoid the need for a new `NetworkNotifiee` struct; although it means the message service has more functions, so maybe not worth it?

I wonder if longer term we should consider refactoring out all the dht/peer discovery logic out of message service struct into it's own struct that can be owned by the message service?

Reply:
> I wonder if longer term we should consider refactoring out all the dht/peer discovery logic out of message service struct into it's own struct that can be owned by the message service?

This seems like a good approach since the message service itself has grown pretty large to support all of the dht-specific methods.